codex/clock.nim

```diff
@@ -13,6 +14,12 @@ method now*(clock: Clock): SecondsSince1970 {.base, upraises: [].} =
 method waitUntil*(clock: Clock, time: SecondsSince1970) {.base, async.} =
   raiseAssert "not implemented"
 
+method start*(clock: Clock) {.base, async.} =
+  discard
+
+method stop*(clock: Clock) {.base, async.} =
+  discard
```

Comment:
Added these, so we could use the base `Clock` type everywhere (and hence the `MockClock`), also in the contract's components. It was needed because of the `OnChainClock` implementation.

Reply:
Is there a reason they're `async`? The `OnChainClock` implementation doesn't seem to need the async.

Reply:
I see now why it is async because of the `await clock.provider.(un)subscribe(...)` calls so, it needs to stay `async`...